### PR TITLE
Provide async control to caller for vm.*Role operations

### PR DESCRIPTION
**Pull request by paulmey**

### CODE REVIEW

File: management/virtualmachine/client.go [hunk 1]
```diff
@@ -99,105 +99,85 @@ func (self VirtualMachineClient) GetRole(cloudserviceName, deploymentName, roleN
...
-	if azureErr != nil {
-		return azureErr
-	}
-
-	return self.client.WaitAsyncOperation(requestId)
+	return self.client.SendAzurePostRequest(requestURL, restartRoleOperationBytes)
 }
 
-func (self VirtualMachineClient) DeleteRole(cloudserviceName, deploymentName, roleName string) error {
+func (self VirtualMachineClient) DeleteRole(cloudserviceName, deploymentName, roleName string) (string, error) {
```
Comment (ahmetalpbalkan) [inline management/virtualmachine/client.go]: hmm shall we maybe introduce a type like `AzureOperationId` to make this obvious?

Comment (paulmey) [inline management/virtualmachine/client.go]: :+1: I'll make another PR for this.

File: management/virtualmachine/client.go [hunk 2]
```diff
@@ -99,105 +99,85 @@ func (self VirtualMachineClient) GetRole(cloudserviceName, deploymentName, roleN
 	return role, nil
 }
 
-func (self VirtualMachineClient) StartRole(cloudserviceName, deploymentName, roleName string) error {
+func (self VirtualMachineClient) StartRole(cloudserviceName, deploymentName, roleName string) (string, error) {
```
Comment (ahmetalpbalkan) [inline management/virtualmachine/client.go]: cloudserviceName is not properly camelCased in this entire file :)

File: management/virtualmachine/client.go (hunk 1 above)
Comment (ahmetalpbalkan) [inline management/virtualmachine/client.go]: @paulmey but that'd require updating signatures of all related methods like WaitAsyncOperation/SendAzureDeleteRequest

Comment (paulmey) [inline management/virtualmachine/client.go]: @ahmetalpbalkan yup, it would... in the whole SDK

File: management/virtualmachine/client.go (hunk 2 above)
Comment (paulmey) [inline management/virtualmachine/client.go]: I'll fix that

**Comment by ahmetalpbalkan** [comment]
LGTM

**Pull request merged**